devices/Button/GpioButton.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Device.Gpio;
+
+namespace Iot.Device.Button
+{
+    /// <summary>
+    /// GPIO implementation of Button.
+    /// Inherits from ButtonBase.
+    /// </summary>
+    public class GpioButton : ButtonBase
+    {
+        private const int DEFAULT_BUTTON_PIN = 37;
+
+        private GpioController _gpioController;
+        private int _buttonPin;
+        private bool _pullUp;
+
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Initialization of the button.
+        /// </summary>
+        /// <param name="buttonPin">GPIO pin of the button.</param>
+        /// <param name="pullUp">If the system is pullup (false = pulldown).</param>
+        /// <param name="doublePressMs">Max ticks between button presses to count as doublepress.</param>
+        /// <param name="holdingMs">Min ms a button is pressed to count as holding.</param>
+        public GpioButton(int buttonPin = DEFAULT_BUTTON_PIN, bool pullUp = true, int doublePressMs = DEFAULT_DOUBLE_PRESS_TICKS, int holdingMs = DEFAULT_HOLDING_MS)
+            : base(doublePressMs, holdingMs)
+        {
+            _gpioController = new GpioController();
+            _buttonPin = buttonPin;
+            _pullUp = pullUp;
+
+            _gpioController.OpenPin(_buttonPin, PinMode.Input);
+            _gpioController.RegisterCallbackForPinValueChangedEvent(_buttonPin, PinEventTypes.Falling | PinEventTypes.Rising, PinStateChanged);
+        }
+
+        /// <summary>
+        /// Handles changes in GPIO pin.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="pinValueChangedEventArgs"></param>
+        private void PinStateChanged(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
+        {
+            switch (GetPinEvent(pinValueChangedEventArgs.ChangeType))
```

Comment:
I'm fine with the current implementation with the 2 functions but I feel like things can be a bit simplify with only one switch. Something like this:
```csharp
switch(pinValueChangedEventArgs.ChangeType)
{
    case PinEventTypes.Falling:
        if( _pullup)
        {
            HandleButtonPressed();
        }
        else
        {
            HandleButtonReleased();
        }
        break;
    case PinEventTypes.Rising:
       if( _pullup)
        {
            HandleButtonReleased();
        }
        else
        {
            HandleButtonPressed();
        }
        break;
}
```

Reply:
Great suggestion, should be fixed now.